packages/components-native/src/Form/types.ts

```diff
@@ -0,0 +1,255 @@
+   */
+  onBeforeSubmit?: (data: FormValues<T>) => Promise<boolean>;
+
+  /**
+   * A callback function that handles the submission of form data
+   */
+  onSubmit: (data: FormValues<T>) => Promise<SubmitResponseType>;
+
+  /**
+   * A callback function that handles any error that occurs during "onSubmit"
+   */
+  onSubmitError: (error: FormErrors) => void;
+
+  /**
+   * A callback function that handles a successful form submission from "onSubmit"
+   */
+  onSubmitSuccess: (data: SubmitResponseType) => void;
+
+  /**
+   * Network or user errors to be displayed as a banner at the top of the form
+   */
+  bannerErrors?: FormBannerErrors;
+
+  /**
+   * Status messages to be displayed as a banner at the top of the form
+   */
+  bannerMessages?: FormBannerMessage[];
+
+  /**
+   * Loading when the initial form data is being fetched
+   */
+  initialLoading?: boolean;
+
+  /**
+   * The initial values of the form inputs
+   * This should be available as soon as initialLoading is set to false
+   */
+  initialValues?: FormValues<DeepPartial<T>>;
+
+  /**
+   * When the validation should happen.
+   * Possible values are "onBlur", "onChange", "onSubmit", "onTouched", and "all".
+   * The default value is "onTouched"
+   */
+  mode?: Mode;
+
+  /**
+   * When the validation after submission should happen.
+   * Possible values are "onBlur", "onChange", and "onSubmit".
+   * The default value is "onChange"
+   */
+  reValidateMode?: Exclude<Mode, "onTouched" | "all">;
+
+  /**
+   * ref object to access react hook form methods and state
+   */
+  formRef?: MutableRefObject<FormRef<T> | undefined>;
+
+  /**
+   * Label to be displayed for the save button
+   */
+  saveButtonLabel?: string;
+
+  /**
+   * @deprecated use `secondaryAction` instead.
+   * Override default save button in the sticky section of the form with another element.
+   */
+  renderStickySection?: (
+    onSubmit: () => void,
+    label: string | undefined,
+    isSubmitting: boolean,
+  ) => JSX.Element;
+
+  /**
+   * Adding a key will save a local copy of the form data that will be used to
+   * recover values when the app is backgrounded or has crashed.
+   */
+  localCacheKey?: string;
+
+  /**
+   * Forms field names that will not be considered for caching.
+   * Useful for omitting sensitive data.
+   */
+  localCacheExclude?: string[];
+
+  /**
+   * A string or array of strings that can be used to identify the pre-filled
+   * data on the form. This can be used to support local caching for forms that
+   * prefill data without inadvertatly applying the cache at the wrong time.
+   *
+   * For example this can be used to when an object is being converted from from
+   * another (Quote being converted into a Job). This will allow the user to
+   * retrieve data from the cache when trying to convert the same object
+   * (same Quote being converted into a Job) following an app crash.
```

Comment:
In my opinion this is fine, looks like it's important for consumers to understand how to use it without crashing the app. Maybe @MichaelParadis has some thoughts too?

Reply:
I think this is fine. Maybe it could be more generalized to talk about how it is form Forms that are dependant on data from another object

```
For example this can be used to when an object is based on data from another object (Quote being converted into a Job). This will allow the user to retrieve data from the cache when trying to create the same object (same Quote being converted into a Job) following an app crash.
```

Reply:
^ I like that

Reply:
Thanks @MichaelParadis ! I changed the comment